src/OMSimulatorLib/System.cpp

```diff
@@ -197,6 +197,9 @@ bool oms3::System::validCref(const oms3::ComRef& cref)
 
 oms_status_enu_t oms3::System::addSubSystem(const oms3::ComRef& cref, oms_system_enu_t type)
 {
+  if(type == oms_system_sc)
+    return logError_NotForScSystem;
+
```

Comment:
It is still the same problem. You have to move this kind of checks into the final iteration, i.e. inside `if (cref.isValidIdent())`.

Reply:
No, in this case it does not matter I think. A strongly coupled system cannot contain subsystems anyway. If it is weakly coupled or TLM, it will still recurse.

Reply:
Ok, I moved it anyway. Even if it doesn't affect the behaviour, there is still no reason to check the types of the parent systems.